url/src/main/scala/uscala/net/URL.scala

```diff
@@ -130,6 +130,7 @@ object Query {
         case a @ Array(key, _*) => key -> a.tail.mkString("=")
       }.groupBy { case (key, _) => key }
        .mapValues(_.toList.map{ case (_, v) => v }.filter(_.nonEmpty))
+       .toMap
```

Comment:
We must call `.toMap` at the end to be safe. In Scala 2.12, `.mapValues` returns a `View` which is a lazy collection which recomputes whenever you use it. This is an insane performance footgun and as a result, Scala 2.13 has changed the type completely so that `MapView` is now returned and is different to `Map` so you are forced into realising you should materialise it using `.toMap` in 99% of cases.

Reply:
Yup! we may also want to change this bit to use `groupMap` at some point

Reply:
Cannot wait for that function, wanted that to be in `Predef` for so long! 👍